java-squid/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -291,14 +345,22 @@ public boolean isEmpty() {
       return false;
     }
 
+    private class Hasher implements PMap.Consumer<Object, Object> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)